tests/test_jobs.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+
+import sky
+from sky import backends
+from sky import exceptions
+from sky import global_user_state
+from sky.utils import db_utils
+
+
+class TestExecutionOnExistingClusters:
+    """Test operations on reserved clusters."""
+
+    @pytest.fixture
+    def _mock_db_conn(self, monkeypatch, tmp_path):
+        tmp_path.mkdir(parents=True, exist_ok=True)
+        db_path = tmp_path / 'state_testing_optimizer_dryrun.db'
+        monkeypatch.setattr(
+            global_user_state, '_DB',
+            db_utils.SQLiteConn(str(db_path), global_user_state.create_table))
+
+    @pytest.fixture
+    def _mock_cluster_state(self, _mock_db_conn, enable_all_clouds):
+        assert 'state.db' not in global_user_state._DB.db_path
+
```

Comment:
Can you write a doctoring or comment that briefly describes the resources that the three clusters have? It'll be very helpful for quickly understanding the test code below.

Reply:
Good point! Added. Thanks!